framework/src/problems/ReferenceResidualProblem.C

```diff
@@ -49,6 +49,16 @@ ReferenceResidualProblem::validParams()
   params.addParam<std::vector<NonlinearVariableName>>(
       "converge_on",
       "If supplied, use only these variables in the individual variable convergence check");
+  params.addParam<bool>(
+      "local_residual_normalization",
+      false,
+      "Flag to optionally perform local normalization of the residual by the reference "
+      "residual dof by dof before the L2 norm is computed");
+  MooseEnum Lnorm("global_L_2 local_L_2 global_L_inf local_L_inf", "global_L_2");
```

Comment:
```suggestion
  MooseEnum Lnorm("global_L2 local_L2 global_Linf local_Linf", "global_L2");
  Lnorm.addDocumentation("global_L2", "Compare the L2 norm of the residual vector to the L2 norm of the absolute reference vector to determine relative convergence");
  Lnorm.addDocumentation("local_L2", "Compute the L2 norm of the residual vector divided componentwise by the absolute reference vector to the L2 norm of the absolute reference vector to determine relative convergence");
  Lnorm.addDocumentation("global_Linf", "");
  Lnorm.addDocumentation("global_L2", "");
```
...

or whatever the right explanation is.